Vagrantfile

```diff
@@ -1,11 +1,57 @@
 Vagrant.configure("2") do |config|
 
-  config.vm.define "controlnode" do |web|
+  # Install required plugins
+  required_plugins = %w( vagrant-hostmanager )
+  plugin_installed = false
+  required_plugins.each do |plugin|
+    unless Vagrant.has_plugin?(plugin)
+      system "vagrant plugin install #{plugin}"
+      plugin_installed = true
+    end
+  end
+
+  # If new plugins installed, restart Vagrant process
+  if plugin_installed === true
+    exec "vagrant #{ARGV.join' '}"
+  end
+  config.vm.box_check_update = true
+  config.vm.graceful_halt_timeout=15
+  config.vbguest.auto_update = false
+  config.ssh.insert_key = false
+  config.ssh.forward_agent = true
+  config.hostmanager.enabled = true
```

Comment:
I'm not 100% sure what the hostmanager does, but I assume it sets a hostname and fills /etc/hosts. The (included) role robertdebock.common does that too.